Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp

```diff
@@ -133,25 +139,61 @@ ExceptionOr<Ref<WebCodecsVideoFrame>> WebCodecsVideoFrame::create(CanvasImageSou
         return initializeFrameFromOtherFrame(videoFrame.releaseNonNull(), WTFMove(init));
     },
 #endif
-    [&] (const RefPtr<HTMLCanvasElement>&) -> ExceptionOr<Ref<WebCodecsVideoFrame>> {
+    [&] (const RefPtr<HTMLCanvasElement>& canvas) -> ExceptionOr<Ref<WebCodecsVideoFrame>> {
         if (!init.timestamp)
             return Exception { TypeError,  "timestamp is not provided"_s };
-        return adoptRef(*new WebCodecsVideoFrame);
+
+        auto videoFrame = canvas->toVideoFrame();
+        if (!videoFrame)
+            return Exception { InvalidStateError,  "Canvas has no frame"_s };
```

Comment:
Should this check the  canvas width and height?

Reply:
Will do